cmd/agent/app/builder.go

```diff
@@ -190,13 +231,29 @@ func (b *Builder) GetProcessors(rep reporter.Reporter, mFactory metrics.Factory)
 	return retMe, nil
 }
 
+func (b *Builder) useTChannelReporter() bool {
+	// if we don't have credentials, we use the tchannel reporter
+	// if we have an auth token or a pair of username+password, we should use the http reporter
+	return b.AuthToken == "" && (b.Username == "" || b.Password == "")
+}
+
 // GetHTTPServer creates an HTTP server that provides sampling strategies and baggage restrictions to client libraries.
-func (c HTTPServerConfiguration) GetHTTPServer(svc string, channel *tchannel.Channel, mFactory metrics.Factory) *http.Server {
-	mgr := httpserver.NewCollectorProxy(svc, channel, mFactory)
-	if c.HostPort == "" {
-		c.HostPort = defaultHTTPServerHostPort
+func (b *Builder) GetHTTPServer(r reporter.Reporter, mFactory metrics.Factory) *http.Server {
+	// TODO: this manager is used for the sampling and baggage restrictions, not sure we need for this here:
+	// is there a non-tchannel sampling/baggage restriction endpoint on the collector side?
+	// for now, we let it be nil for non-TChannel reporters
```

Comment:
not sure how it would work then, when clients request their configs, the nil mgr will cause a panic

Reply:
What should be done then? Should I just implement a dummy endpoint, returning some default data?

Reply:
That seems like the weak spot of this PR - we're trying to hack just one flow between the agent and collector where in practice we need to replace the whole API with TLS-capable transport. Maybe we shouldn't do it piecemeal and just Do The Right Thing by implementing gRPC.

Reply:
I might have misunderstood this back in November. Based on the last meeting, looks like we want gRPC *in addition* to this, but then, the problem reported on this comment would still exist, wouldn't it?